routing_common/transit_serdes.hpp

```diff
@@ -40,14 +40,34 @@ class Serializer
 public:
   Serializer(Sink & sink) : m_sink(sink) {}
 
-  template<typename T>
-  typename std::enable_if<std::is_integral<T>::value || std::is_enum<T>::value>::type
-      operator()(T const & t, char const * /* name */ = nullptr)
+  template <typename T>
+  typename std::enable_if<(std::is_integral<T>::value || std::is_enum<T>::value) &&
+                          !std::is_same<T, uint32_t>::value && !std::is_same<T, uint64_t>::value &&
+                          !std::is_same<T, int32_t>::value &&
+                          !std::is_same<T, int64_t>::value>::type
+  operator()(T const & t, char const * /* name */ = nullptr)
   {
     WriteToSink(m_sink, t);
   }
 
-  void operator()(double d, char const * name = nullptr)
+  template <typename T>
+  typename std::enable_if<std::is_same<T, uint32_t>::value ||
```

Comment:
Why not to have VisitVarUint() methods in visitor and get rid of these ugly enable_ifs?

Reply:
I'd prefer leave it as is. I like that I should not point at type of value (by typing method) when I declare the visitors.